### DISCUSSION TITLE
Struggling with MongoDB Aggregate code formatting

### DISCUSSION THREAD

**Discussion by dan-burt** [Budibase/budibase#8537] [Help] [answered]
I am trying to replicate the existing functionality of a PHP-based website. This uses various queries and aggregate pipelines.

I am trying to add Stages to the "pipeline" version. And this is what I have so far...

<img width="922" alt="image" src="https://user-images.githubusercontent.com/29703514/200161978-8750d4af-6366-455e-a6b1-7505b2764784.png">

The error is in the red box at the bottom.

I have tried using square brackets, curly brackets, around the whole block.

Sure this is a simple syntax thing... And seeing it once would likely allow me to apply to the other constructs ($min, $max, $first, etc)

**Comment by melohagan** [comment, marked as answer]
Hey @dan-burt 

I think it is just a syntax issue. So for example: https://www.mongodb.com/docs/manual/reference/operator/aggregation/group/#count-the-number-of-documents-in-a-collection

`$group`
```js
  {
     "totalSaleAmount": { "$sum:" { "$multiply": [ "$price", "$quantity" ] } },
     "averageQuantity": { "$avg": "$quantity" },
     "count": { "$sum": 1 }
  }
```

**Reply by dan-burt** [reply]
Yes, this was formatting.

I just kept trying different quotes, etc...

This code ended up working, in my use case:

Stage 1 ("$match", also using a Binding):
```
{
	"compId": "ObjectId('{{id}}')"
}
```

Stage 2 ($group):
```
{
  "_id": ["$playerId", "$playerName"],
	"score": {"$sum": "$computedScore"},
	"thru": {"$sum": "$computedThru"},
	"hcapHigh": {"$max": "$hcapPlaying"},
	"hcapLow": {"$min": "$hcapPlaying"}
}
```

NOTE - square or curly braces as well...

**Reply by melohagan** [reply]
Awesome, thanks for sharing what worked for you!
